src/sota_tools/ostree_http_repo.cc

```diff
@@ -38,20 +35,16 @@ OSTreeRef OSTreeHttpRepo::GetRef(const std::string &refname) const { return OSTr
 
 bool OSTreeHttpRepo::FetchObject(const boost::filesystem::path &path) const {
   CURLcode err = CURLE_OK;
-  CurlEasyWrapper easy_handle;
-  curlEasySetoptWrapper(easy_handle.get(), CURLOPT_VERBOSE, get_curlopt_verbose());
-  server_->InjectIntoCurl(path.string(), easy_handle.get());
-  curlEasySetoptWrapper(easy_handle.get(), CURLOPT_WRITEFUNCTION, &OSTreeHttpRepo::curl_handle_write);
+  server_->InjectIntoCurl(path.string(), easy_handle_.get());
```

Comment:
At first glance, it looks like the auth creds/certs are set for each request while it can be set once in the curl context and then just `CURLOPT_URL` needs to be set for each request for an object so TCC/TLS connection is reused.

Reply:
I suppose the goal here is to make sure that TCP/TLS connection is reused for each request for both the downloading and pushing objects to Treehub. I suppose libcurl should do it by default as long as "_Then reusing the same easy handle will ensure it can reuse its connections.", but I am not sure if it works if we setup the creds in the "easy handle" for each request.

Reply:
I did wonder if I should refactor that function, and there is room for some minor optimization there, but it seems like the connection is getting reused without messing with it, so I didn't change it.

Reply:
+1: